cmd/juju/secrets/grantrevoke.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package secrets
+
+import (
+	"strings"
+
+	"github.com/juju/cmd/v3"
+	"github.com/juju/errors"
+
+	apisecrets "github.com/juju/juju/api/client/secrets"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/core/secrets"
+)
+
+type grantSecretCommand struct {
+	modelcmd.ModelCommandBase
+
+	secretURI *secrets.URI
+	apps      []string
+
+	secretsAPIFunc func() (GrantRevokeSecretsAPI, error)
+}
+
+// GrantRevokeSecretsAPI is the secrets client API.
+type GrantRevokeSecretsAPI interface {
+	GrantSecret(*secrets.URI, []string) ([]error, error)
+	RevokeSecret(*secrets.URI, []string) ([]error, error)
+	Close() error
+}
+
+// NewGrantSecretCommand returns a command to grant view access of a secret to applications.
+func NewGrantSecretCommand() cmd.Command {
+	c := &grantSecretCommand{}
+	c.secretsAPIFunc = c.secretsAPI
+	return modelcmd.Wrap(c)
+}
+
+func (c *grantSecretCommand) secretsAPI() (GrantRevokeSecretsAPI, error) {
+	root, err := c.NewAPIRoot()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return apisecrets.NewClient(root), nil
+}
+
+// Info implements cmd.Command.
+func (c *grantSecretCommand) Info() *cmd.Info {
+	doc := `
+Grant access to view the value of a specified secret.
+Access is granted in the appliation scope.
```

Comment:
This can be simplified for the user as 

Grant an application access to view the value of a secret.